packages/language-server/src/core/parseJS.ts

```diff
@@ -17,44 +17,11 @@ export function extractScriptTags(
 	htmlDocument: HTMLDocument,
 	ast: ParseResult['ast']
 ): VirtualFile[] {
-	const embeddedJSFiles: VirtualFile['embeddedFiles'] = [];
-	for (const [index, root] of htmlDocument.roots.entries()) {
-		if (
-			root.tag === 'script' &&
-			root.startTagEnd !== undefined &&
-			root.endTagStart !== undefined &&
-			isIsolatedScriptTag(root)
-		) {
-			const scriptText = snapshot.getText(root.startTagEnd, root.endTagStart);
-
-			embeddedJSFiles.push({
-				fileName: fileName + `.${index}.mts`,
```

Comment:
I am not quite sure if this is the right approach. Should there be some sort of level visible in the filename? E.g. `Header.astro.0.mts` (root level), but if the root level would have some children `Header.astro.0.0.mts` (first child).